agr_api/src/main/java/org/alliancegenome/api/service/AffectedGenomicModelESService.java

```diff
@@ -39,8 +48,37 @@ public JsonResultResponse<AffectedGenomicModelDocument> getGeneModels(
 		if (sortFields == null) {
 			sortFields = sortingSetMap.get("default");
 		}
-		sortFields.forEach(sortField -> sortingMap.put(sortField, SortOrder.ASC));
-		SearchResponse searchResponse = getSearchResponse(query, pagination, sortingMap, null, debug);
+
+		// Create search source builder for custom script sorting
+		SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+		searchSourceBuilder.query(query);
+		searchSourceBuilder.from(pagination.getStart());
+		searchSourceBuilder.size(pagination.getLimit());
+
+		// Handle sorting with case-insensitive script for model name field
+		for (String sortField : sortFields) {
+			if ("model.agmFullName.formatText.keyword".equals(sortField)) {
+				// Use script for case-insensitive sorting
+				Script script = new Script(ScriptType.INLINE, "painless",
+					"if (doc['model.agmFullName.formatText.keyword'].size() > 0) { "
+						+ "doc['model.agmFullName.formatText.keyword'].value.toLowerCase() } else { '' }",
+					Collections.emptyMap());
+				ScriptSortBuilder scriptSort = SortBuilders.scriptSort(script, ScriptSortBuilder.ScriptSortType.STRING);
+				scriptSort.order(SortOrder.ASC);
+				searchSourceBuilder.sort(scriptSort);
+			} else {
+				searchSourceBuilder.sort(sortField, SortOrder.ASC);
+			}
+		}
+
+		SearchRequest searchRequest = new SearchRequest(ConfigHelper.getEsIndex());
+		searchRequest.source(searchSourceBuilder);
+		SearchResponse searchResponse;
+		try {
+			searchResponse = EsClientFactory.getDefaultEsClient().search(searchRequest, RequestOptions.DEFAULT);
+		} catch (Exception e) {
+			throw new RuntimeException("Error executing search", e);
+		}
```

Comment:
@cmpich on the field in the mapping you can add the .sort() method and then you can just sort on the sort field. As in `searchSourceBuilder.sort("model.agmFullName.formatText.sort", SortOrder.ASC);`

Reply:
```java -- Mapping.java:325
		if (sort) {
			//buildProperty("sort", "keyword", null, null, "lowercase");
			buildProperty("sort", "keyword", null, null, "smart_alpha_sort");
		}
```
As the smart_alpha doesn't work we might need to revert this change.